rfcs/2021-11-22-design-tokens.md

```diff
@@ -0,0 +1,106 @@
+
+By deciding on a [Single Source Of Truth](https://en.wikipedia.org/wiki/Single_source_of_truth) (SSOT) for our design rules, we can organize and transform them for any system which utilizes them.
+
+## Benefits
+
+1. **Consistency** - Between design and code, utilizing a single set of variables everywhere allows a design proof to accurately describe the visual representation of a component as a set of variables. This provides better parity between what's designed and what is seen in the rendered usage on all platforms.
+
+   Looking at the [CSSStats for healthcare.gov/see-plans](https://cssstats.com/stats?url=healthcare.gov/see-plans), there are currently 29 color variations, many of which are nearly identical. Utilizing design tokens could bring this number down considerably and provide better brand consistency.
+
+2. **Organization** - By starting from the atomic level, we can use naming conventions for our tokens that are simple and specific to the variable they represent. By keeping these tokens separate from the systems they are utilized in, they can be managed as a discrete unit. Agreement on naming conventions of variables could eventually create parity between Sketch libraries, DS Sass variables, and future design and development tooling.
+3. **Interoperability** - Design tokens allow a huge amount of flexibility for both moving a design system to a new platform, or creating a new design system. Should an iOS/Android app ever be built, a new web application created, etc. That system would have a framework ready to utilize for its theming which would be consistent with the other brand applications.
+4. **Opportunity for theming** - Decoupling design rules from the codebase allowing the transposition-ing of these static design decisions in various ways, i.e. theming/high contrast. By replacing all tokens in a system with new tokens you can essentially theme swap, assuming the consuming system is utilizing tokens at all levels appropriately.
+
+## Risks
+
+1. There can be a reductive quality to design tokens in that there might be a more limited set of colors/spacing options/etc.. to work with for the consuming applications. Someone might have to sell consuming systems on the advantages to brand consistency, which they hopefully already understand.
+
+## Scope/Outline
+
+**An outline for how this process could proceed:**
+
+1. The SSOT should be decided on, at the moment I see two options:
+
+   1. **The DSM becomes the source of truth**. Designers and developers work together to consolidate the current set of design rules residing in the codebase/Sketch and place them into the DSM. In this case, with that work done, an application can be written in the design system to utilize the [DSM Token API](https://support.invisionapp.com/hc/en-us/articles/360049950931) to periodically update it's set of Sass variables with data from these design tokens and store them in a new NPM package.
```

Comment:
Does the DSM's design tokens system have some sort of approval process?

Reply:
Maybe @line47 or @ericwbailey  can answer that? I don't believe the DSM's system offers any advantages over storing these in our own package/repo with git versioning.

Reply:
The DSM doesn't have much in terms of an approval process things are either in a draft state or published. I guess you could review things in draft and then publish via the DSM.